C#/C#.sublime-syntax

```diff
@@ -1,276 +1,1593 @@
+      set: [method_accessor, method_params_bracket]
+    - match: '(?:(?#<=>)|{{name}})(\.)'
+      captures:
+        0: meta.method.identifier
+        1: entity.other.inherited-class.cs
+        2: punctuation.accessor.dot.cs
+    - match: '{{name}}'
+      scope: entity.name.variable.property.cs
+    - match: '(?=\{)'
+      set: method_accessor
+
+  method_params:
+    - meta_scope: meta.method.identifier
+    - match: \)
+      scope: punctuation.definition.parameters.end.cs
+      pop: true
+    - match: (?=\S)
+      push: [method_param, method_param_type]
+
+  method_params_bracket:
+    - meta_scope: meta.method.identifier
+    - match: \]
+      scope: punctuation.definition.parameters.end.cs
+      pop: true
+    - match: (?=\S)
+      push: [method_param, method_param_type]
+
+  method_param:
+    - match: '{{name}}'
+      scope: variable.parameter.cs
+    - match: '='
+      scope: punctuation.separator.definition.default-value.cs
+      push: line_of_code_in
+    - match: ','
+      scope: punctuation.separator.parameter.function.cs
+      pop: true
+    - match: (?=\}|\)|>|\]|;)
+      pop: true
+
+  method_param_type:
+    - match: '\s*(out|ref|this|params)\s+'
+      scope: storage.modifier.parameter.cs
+    - match: \s
+      pop: true
+    - include: type
+    - include: tag
+
+  method_accessor:
+    - meta_scope: meta.method.property.cs
+    - match: \{
+      scope: punctuation.section.function.begin.cs
+      set:
+        - match: \}
+          scope: punctuation.section.function.end.cs
+          set:
+            - match: =
               scope: keyword.operator.assignment.cs
-              push:
-                - match: ';'
-                  pop: true
-                - include: code
-            - include: code
-  code:
-    - include: block
-    - include: comments
-    - include: class
-    - include: constants
-    - include: storage-modifiers
+              set: line_of_code_in
+            - match: (?=\S)
+              pop: true
+        - match: \b(get)\b
+          scope: storage.type.function.accessor.get.cs
+          push: method_body
+        - match: \b(set)\b
+          scope: storage.type.function.accessor.set.cs
+          push: method_body
+        - match: '{{visibility}}'
+          scope: storage.modifier.access.cs
+        - include: tag
+    - match: '=>'
+      scope: punctuation.section.function.begin.cs
+      set: line_of_code_in
+    - match: \S
+      scope: invalid.illegal
+      pop: true
+
+  tag:
+    - match: (\[)\s*(assembly|module|field|event|method|param|property|return|type)\s*(:)
+      captures:
+        1: punctuation.definition.tag.begin.cs
+        2: storage.modifier.target.cs
+        3: punctuation.separator.key-value.tag.cs
+      push: tag_in
+    - match: \[
+      scope: punctuation.definition.tag.begin.cs
+      push: tag_in
+
+  tag_in:
+    - meta_scope: meta.attribute.cs
+    - match: '{{name}}(\()'
+      captures:
+        1: entity.name.tag
```

Comment:
I'm not sure I agree with the scope `entity.name.tag` here. Perhaps it should just be `variable.other.type.attribute.cs` or something instead?